test/unit/matchers.py

```diff
@@ -0,0 +1,31 @@
+from pathlib import Path
+
+
+class UsedLogPath:
+    def __init__(self, log_path: Path, task_input_parameter):
+            self.log_path = log_path
+            self.task_input_parameter = task_input_parameter
+
+    def __repr__(self):
+        return f"UsedLogPath({str(self.log_path)}, {self.task_input_parameter})"
```

Comment:
The representation should be designed in a way, that you can potentially diff them, because they are used in diff in case of a failing assert